primap2/csg/_compose.py

```diff
@@ -136,7 +138,25 @@ def compose(
         if pbar is not None:
             pbar.close()
 
-    return xr.Dataset(result_das).pr.quantify()
+    result_ds = xr.Dataset(result_das).pr.quantify()
+    # composing removes the priority dimensions, also remove the attrs describing
+    # the priority dimensions
+    result_ds.attrs = input_data.attrs.copy()
+    for dim_key in priority_dimensions:
+        if isinstance(dim_key, str) and "(" in dim_key:
+            dim = dim_key.split("(", 1)[0][:-1]
+        else:
+            dim = dim_key
+        if dim == "area":
+            del result_ds.attrs["area"]
+        elif dim == "category":
+            del result_ds.attrs["cat"]
+        elif dim == "scenario":
+            del result_ds.attrs["scen"]
+        elif dim not in ("provenance", "model", "source"):
+            result_ds.attrs["sec_cats"].remove(dim_key)
```

Comment:
The dataset I use is actually not a valid primap2 dataset as it doesn't contain the source dimension. So maybe in this case the problem is that I can save and load it despite it being invalid.

Reply:
Generally, checking a dataset and loading/saving it are different operations, and I would like to keep the possibility to save and load invalid datasets.

Reply:
So, I'd say: yes, let's be more lenient in what we accept in `compose` and only remove the `sec_cats` entry if it exists. But for saving/loading we should either keep it as is or maybe add an option to do the verification instead of doing it always.